mobify/platform-scaffold
# [Integration Manager] Command audit

This PR makes all IM command parameters explicit as well as fixes some outstanding bugs related to that. 

 **JIRA**: [**WEBDATA-17**: Finalize Command API](https://mobify.atlassian.net/browse/WEBDATA-17)
 **Linked PRs**: N/A

## Changes
- Removes the `...args` parameter style for all IM commands and replaces them with the actual list of arguments each command expects
- Adds JSDocs to every IM command
- Centralizes the `XYZ_FORM_NAME` constants as they are used by both selectors and `/app/containers/...`
- Renamed all `getXyxPage` commands to `initXyzPage` as not all commands need to fetch data

## How to test-drive this PR
- Run `npm run dev`
- Make sure `main.jsx` has the Merlin's connector configured
    ```js
    import connector from './integration-manager/_merlins-connector'
    import {registerConnector} from './integration-manager'
    ````
- Preview [Magento Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- **Verify**: Make sure all features work (basic smoke test through checking and `npm run test:e2e` should be enough
- Switch `main.jsx` to use the SFCC connector
    ```js
    import connector from './integration-manager/_demandware-connector'
    import {registerConnector} from './integration-manager'
    ````
- Preview [SFCC Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fmobify-tech-prtnr-na03-dw.demandware.net%2Fon%2Fdemandware.store%2FSites-2017refresh-Site%2Fdefault%2FHome-Show&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- **Verify**: Make sure all features work (basic smoke test through checking and `npm run test:e2e` should be enough
